process/smartContract/process.go

```diff
@@ -1127,21 +1034,27 @@ func (sc *scProcessor) ProcessSmartContractResult(scr *smartContractResult.Smart
 }
 
 func (sc *scProcessor) processSimpleSCR(
-	scr *smartContractResult.SmartContractResult,
-	dstAcc state.UserAccountHandler,
+	scResult *smartContractResult.SmartContractResult,
+	destinationScAccount state.UserAccountHandler,
 ) error {
-	err := sc.updateSmartContractCode(dstAcc, &vmcommon.OutputAccount{Code: scr.Code, Address: scr.RcvAddr}, scr)
+	outputAccount := &vmcommon.OutputAccount{
+		Code:         scResult.Code,
+		CodeMetadata: scResult.CodeMetadata,
+		Address:      scResult.RcvAddr,
+	}
+
+	err := sc.updateSmartContractCode(destinationScAccount, outputAccount, scResult)
 	if err != nil {
 		return err
 	}
 
-	if scr.Value != nil {
-		err = dstAcc.AddToBalance(scr.Value)
+	if scResult.Value != nil {
+		err = destinationScAccount.AddToBalance(scResult.Value)
 		if err != nil {
 			return err
 		}
 
-		err = sc.accounts.SaveAccount(dstAcc)
+		err = sc.accounts.SaveAccount(destinationScAccount)
```

Comment:
There might be a problem here, from previous refactoring. The `destinationScAccount` is modified in `sc.updateSmartContractCode()`, but it is saved only if `scResult.Value != nil`. Is there a case when `scResult.Value == nil`, but the account is still modified in `sc.updateSmartContractCode()`? @sasurobert @andreibancioiu @camilbancioiu

Reply:
Nice catch! Fixed to always call `SaveAccount` in `processSimpleSCR`. Ok?